dbms/src/Storages/MergeTree/ReplicatedMergeTreeRestartingThread.cpp

```diff
@@ -28,134 +28,141 @@ namespace ErrorCodes
     extern const int REPLICA_IS_ALREADY_ACTIVE;
 }
 
+namespace
+{
+    constexpr auto retry_period_ms = 10 * 1000;
+}
 
 /// Used to check whether it's us who set node `is_active`, or not.
 static String generateActiveNodeIdentifier()
 {
     return "pid: " + toString(getpid()) + ", random: " + toString(randomSeed());
 }
 
-
 ReplicatedMergeTreeRestartingThread::ReplicatedMergeTreeRestartingThread(StorageReplicatedMergeTree & storage_)
     : storage(storage_),
     log(&Logger::get(storage.database_name + "." + storage.table_name + " (StorageReplicatedMergeTree, RestartingThread)")),
-    active_node_identifier(generateActiveNodeIdentifier()),
-    thread([this] { run(); })
+    active_node_identifier(generateActiveNodeIdentifier())
 {
+    check_period_ms = storage.data.settings.zookeeper_session_expiration_check_period.totalSeconds() * 1000;
+
+    /// Periodicity of checking lag of replica.
+    if (check_period_ms > static_cast<Int64>(storage.data.settings.check_delay_period) * 1000)
+        check_period_ms = storage.data.settings.check_delay_period * 1000;
+
+    storage.queue_updating_task_handle = storage.context.getSchedulePool().addTask("queue_updating_task_handle", [this]{ storage.queueUpdatingThread(); });
```

Comment:
The task is inconsistently named (after the variable name, not the function name).

Reply:
`StorageReplicatedMergeTree::queueUpdatingTask` (possible with the table name)

Reply:
OK this is clear!

Reply:
Fixed in commit 6629b03af9bb4ebf50b2ab94b7c3a965b773907c